app/containers/GuideProfilePage/index.js

```diff
@@ -57,17 +45,22 @@ export class GuideProfilePage extends React.Component { // eslint-disable-line r
 
               <div className="d-lg-none">
                 {/* Mobile/tablet view */}
-                {/* Currently using dummy profile */}
-                <ProfileHeader guide {...sampleProfile} />
+                {
+                  this.props.guideData.name !== 'LOADING' &&
```

Comment:
We shouldn't use the name "loading" as a switch. I did that just for fun to see what it looked like. Instead, if we need to switch on "are we in a loading state?", there is a specific state variable for that

https://github.com/Octember/the-approach/blob/master/app/containers/GuideProfilePage/reducer.js#L27

In the future, we can have a very slick page load that looks OK while still waiting for the API response. For now, I think it's ok to ignore it mostly, but keeping it in the back of our minds.

If you want to do that here, make a new selector for this container `selectIsLoading` or something, plug it through here, and use `this.props.isLoading` instead.

I don't care much either way what you do here, but we should not depend on that magic string 'LOADING'

Reply:
Made changes that are hopefully acceptable. This was kind of good practice with selectors, but I found it frustrating as there isn't a straightforward way to access STATE_LOADING before getting guide data (maybe I'm missing something). I also don't understand STATE_LOADING being a constant when it's value can be changed (maybe I don't understand how the constants are used in the context of a React-Redux project).

Reply:
Yeah so two things;

`STATE_LOADING` is just a key. it's value is always "GuideProfilePage/loading_state" or something. The key is used to access the value of whether or not the state is loading or not. In the reducer file, you can see we set the value of the state at that key when we make the API call, and after the API call finishes

Then in the selectors, we read from the state with the same key to get that value.

I agree there is some weirdness here. If we're still loading, certain values won't be set. We can work on this down the line, for now I'm OK with it being somewhat of a grey area. Loading state is a cherry on top of the page load flow; once the API is sufficiently fast it will be a very short period anyway (50ms or so)